src/main/java/com/direwolf20/buildinggadgets/common/world_save/TemplateWorldSave.java

```diff
@@ -0,0 +1,77 @@
+package com.direwolf20.buildinggadgets.common.world_save;
+
+import com.direwolf20.buildinggadgets.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.schema.template.Template;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.INBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraft.world.storage.WorldSavedData;
+import net.minecraftforge.common.util.Constants.NBT;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public final class TemplateWorldSave extends WorldSavedData {
+    public static TemplateWorldSave getInstance(ServerWorld world) {
+        return null;//TODO: evaluate correct load Method
+    }
+
+    private static final String NAME = BuildingGadgets.MOD_ID + "_template_data";
+    private static final String KEY_TEMPLATE_LIST = "templates";
+    private static final String KEY_ID = "id";
+    private static final String KEY_DATA = "data";
+    private final Map<UUID, Template> loadedTemplates;
+
+    public TemplateWorldSave() {
+        super(NAME);
+        this.loadedTemplates = new HashMap<>();
+    }
+
+    @Override
+    public void read(CompoundNBT nbt) {
+        if (! nbt.contains(KEY_TEMPLATE_LIST, NBT.TAG_LIST)) {
+            BuildingGadgets.LOGGER.warn("No templates found to load from nbt data.");
+            return;
+        }
+        ListNBT templates = nbt.getList(KEY_TEMPLATE_LIST, NBT.TAG_COMPOUND);
+        BuildingGadgets.LOGGER.debug("Loading {} template{}.", templates.size(), getSExtension(templates.size()));
+        int loadCount = 0;
+        for (INBT templateNBT : templates) {
+            CompoundNBT templateCompound = (CompoundNBT) templateNBT;
+            UUID id = templateCompound.getUniqueId(KEY_ID);
+            if (Template.deserializeNBT(templateCompound.getCompound(KEY_DATA))
```

Comment:
Oh jesus no. We both know why I'm saying this. This isn't up for discussion. Do not write code like this.

Reply:
Java 9 adds ifpresent or else... So how else am I supposed to write this?

Reply:
If we were in kotlin I'd immediately write an extension function, but having a static Method do that is kind of awkward.

Reply:
The non-disgusting way. You don't dump a lambda into a if statement.

Reply:
> You don't dump a lambda into a if statement.

I really have no Problem with that... 

So if that's you're problem, then I' going to assign it to a boolean variable

Reply:
When we're continually talking about code readability and cleanliness, we don't mean what ever that is above. Make your code verbose so it's easier to read. The same with the lack of newlines between logical code groups. You're just dumping huge code blocks into a otherwise well formatted codebase.

Reply:
To the formatting: as I said on Discord it is very unfinished. I'll reformat just before I think it is ready for another review.

Also as I've got no trouble reading this, I kind of don't understand the Problem, even though I'll change it. In fact I find that quite elegant as long as there's no ifPresentOrElse

Reply:
Again, it's really not up for discussion. We have differing options on readability clearly. I hope to see formatting throughout the PR upon a proper review request.

Reply:
Kind of sad, that we can't all agree on everything. Though it does produce productive discussions.